portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,125 @@
+		answers=['San Francisco, CA', 'Vancouver, BC', 'Toronto, ON', 'Palo Alto CA'], correctAnswerIndex=1));
+questionsArray.push(
+    new Question(question='Where do I attend university?',
+		answers=['University of British Columbia', 'Simon Fraser University',
+		  'Stanford University', 'New York University'], correctAnswerIndex=0));
+questionsArray.push(
+    new Question(question='In which of these classes did I receive the highest grade?',
+		answers=['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], correctAnswerIndex=1));
+questionsArray.push(
+    new Question(question='In which of these classes did I receive the lowest grade?',
+		answers=['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], correctAnswerIndex=3));
+
+/**
+ * Display the questions on the webpage, using dropdown boxes for the answers
+ */
 function presentQuiz() {
-  // create array of quiz questions 
-  const questions;
+  const questionsContainer = document.getElementById('questions-container');
+  for (let questionNum = 0; questionNum < questionsArray.length; questionNum++) {
+    const question = questionsArray[questionNum];
+    const answers = question.getAnswers();
+    const correctIndex = question.getCorrectAnswerIndex();
+    // Display the options
+    const questionStatementElement = document.createElement('p');
+    const questionStatementText = document.createTextNode(`Question ${questionNum}: ${question.getQuestion()}`);
+    questionStatementElement.appendChild(questionStatementText);
 
-  const quizContainer = document.getElementById('quiz-container');
-  const questionContainer = document.getElementById('question-container');
-  const answersContainer = document.getElementById('answers-container');
-  const submitContainer = document.getElementById('submit-container');
+    const quizArea = document.createElement('select');
+    quizArea.setAttribute('id', generateId(questionNum));
+    for (let answerIndex = 0; answerIndex < answers.length; answerIndex++) {
+      const option = document.createElement('option');
+      option.setAttribute('value', answerIndex);
+      const answer = document.createTextNode(answers[answerIndex]);
+      option.appendChild(answer);
+      quizArea.appendChild(option);
+    }
 
-  for (q in questions) {
-    // display the question in a container
-    questionContainer.innerText = q.getQuestion();
-    // Display the options
-    // Use radio buttons
-    // Have a submit button
-    // When user clicks submit button display correct answer
+    const emptyAnswer = document.createElement('option');
+    emptyAnswer.setAttribute('selected', '');
+    emptyAnswer.setAttribute('value', '');
+    quizArea.appendChild(emptyAnswer);
+
+    questionsContainer.appendChild(questionStatement);
+    questionsContainer.appendChild(quizArea);
```

Comment:
This function is doing a lot - it might make sense to break it into smaller functions s.t. the function names reflect each chunk of logic that's being invoked: "displayOptions" might b a good one, along with addPossibleAnswer, or something like that.

Reply:
Ok, I'll refactor further.